src/anndata/_io/specs/methods.py

```diff
@@ -506,10 +511,12 @@ def write_vlen_string_array_zarr(
 ):
     import numcodecs
 
-    # Workaround for https://github.com/zarr-developers/numcodecs/issues/514
-    # TODO: Warn to upgrade numcodecs if fixed
-    if not elem.flags.writeable:
-        elem = elem.copy()
+    if Version(numcodecs.__version__) < Version("0.13"):
+        msg = "Old numcodecs version detected. Please update for improved performance and stability."
+        warnings.warn(msg)
+        # Workaround for https://github.com/zarr-developers/numcodecs/issues/514
+        if hasattr(elem, "flags") and not elem.flags.writeable:
+            elem = elem.copy()
```

Comment:
Huh. Seems like coverage has been broken for a while, why are we getting anything?

Reply:
It looks like 3.9 is publishing?

Reply:
For potential future people seeing this: only the mindeps and 3.12 jobs are publishing.

The reason it works is because we’re publishing twice: once to azure, once to codecov.io. The second one worked all along.